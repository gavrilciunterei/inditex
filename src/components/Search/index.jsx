import React from 'react';
import { SearchIcon } from '../SvgIcons';

function Search({ name, action }) {
  return (
    <div className="flex  mb-6  justify-center sm:justify-end">
      <div className="flex relative mx-automax-w-sm ">
        <input
          className="border-2 border-primary bg-red transition h-12 px-5 pr-16 rounded-md focus:outline-none w-full text-black text-lg "
          type="search"
          name="search"
          placeholder={name}
          onChange={(event) => action(event.target.value)}
        />
        <div type="submit" className="absolute right-2 top-3 mr-4">
          <SearchIcon />
        </div>
      </div>
    </div>
  );
}

export default Search;
