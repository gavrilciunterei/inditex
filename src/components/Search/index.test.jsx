import React from 'react';
import renderer from 'react-test-renderer';
import Search from './index';
import { render, screen, fireEvent } from '@testing-library/react';

it('Test Search', () => {
  const tree = renderer
    .create(<Search name="search" action={() => true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test Write input', () => {
  render(<Search name="search" action={() => true} />);
  const input = screen.getByPlaceholderText('search');
  fireEvent.change(input, { target: { value: 'Acer' } });
});
