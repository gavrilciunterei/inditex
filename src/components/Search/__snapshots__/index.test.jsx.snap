// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search 1`] = `
<div
  className="flex  mb-6  justify-center sm:justify-end"
>
  <div
    className="flex relative mx-automax-w-sm "
  >
    <input
      className="border-2 border-primary bg-red transition h-12 px-5 pr-16 rounded-md focus:outline-none w-full text-black text-lg "
      name="search"
      onChange={[Function]}
      placeholder="search"
      type="search"
    />
    <div
      className="absolute right-2 top-3 mr-4"
      type="submit"
    >
      <svg
        className=" h-6 w-6 "
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
